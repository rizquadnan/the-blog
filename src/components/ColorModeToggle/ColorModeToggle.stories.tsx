import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { ColorModeToggle } from './ColorModeToggle'

type TColorModeToggle = React.ComponentProps<typeof ColorModeToggle>
export default {
  title: 'UserInputs/ColorModeToggle',
  component: ColorModeToggle,
} as Meta<TColorModeToggle>

const Template: Story<TColorModeToggle> = () => <ColorModeToggle />

export const Default = Template.bind({})
