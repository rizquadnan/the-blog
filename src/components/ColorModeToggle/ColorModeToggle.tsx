import { MoonIcon, SunIcon } from '@chakra-ui/icons'
import { Flex, IconButton, useColorMode } from '@chakra-ui/react'

export function ColorModeToggle() {
  const { colorMode, toggleColorMode } = useColorMode()

  return (
    <Flex justifyContent="flex-end" padding="24px" position="fixed" right="0">
      {colorMode === 'dark' ? (
        <IconButton
          aria-label="Change to light mode"
          icon={<SunIcon />}
          onClick={toggleColorMode}
        />
      ) : (
        <IconButton
          aria-label="Change to dark mode"
          icon={<MoonIcon />}
          onClick={toggleColorMode}
        />
      )}
    </Flex>
  )
}
