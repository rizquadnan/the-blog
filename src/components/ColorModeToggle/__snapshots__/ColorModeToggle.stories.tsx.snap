// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInputs/ColorModeToggle Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Change to dark mode",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "userinputs-colormodetoggle--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
